### PR TITLE
Issue #3660: removed checkstyle_packages.xml from main

### PR DESCRIPTION
**Pull request by rnveach**
Issue #3660

### CODE REVIEW

File: config/pmd-test.xml [hunk 1]
```diff
@@ -85,6 +86,7 @@
             | /ClassOrInterfaceDeclaration[@Image='AstRegressionTest']//MethodDeclarator[@Image='testImpossibleExceptions']
             | //ClassOrInterfaceDeclaration[@Image='AstRegressionTest']//MethodDeclarator[@Image='testImpossibleValid']
             | //ClassOrInterfaceDeclaration[@Image='AllChecksTest']//MethodDeclarator[@Image='testAllModulesAreReferencedInConfigFile']
+            | //ClassOrInterfaceDeclaration[@Image='PackageNamesLoaderTest']//MethodDeclarator[@Image='testPackagesFile']
```
Comment (romani) [inline config/pmd-test.xml]: @rnveach , please inline `validatePackageNames` to testPackagesFile and this suppression will not be required.

**Comment by rnveach** [comment]
@romani 
> do we really need support of checkstyle_packages.xml - https://github.com/checkstyle/checkstyle/blob/master/src/xdocs/config.xml#L761 ?

We use this file to specify which packages to scan for 3rd party modules at https://github.com/checkstyle/checkstyle/blob/46fa8090954480eecdeb6e7367f0e30655853f42/src/main/java/com/puppycrawl/tools/checkstyle/PackageNamesLoader.java#L137-L147 .

If you are asking why don't we pull in all modules in the full classpath instead of restricting to a list of packages from users, than the only good reasons I have are:
* Error message will be slightly more vague as it won't say what packages we were looking at. See: https://github.com/checkstyle/checkstyle/pull/4886 which we just updated recently
* Scanning the entire classpath could take longer as we will be looking at more classes and not restricting any by package.

I will look into the second point.

**Comment by romani** [comment]
@rnveach , please rebase to resolve conflict.

> Scanning the entire classpath could take longer 

it could be too much, it still good to have from user package to scan.

File: config/pmd-test.xml (hunk 1 above)
Comment (rnveach) [inline config/pmd-test.xml]: Done.

**Comment by rnveach** [comment]
Rebased.

**Pull request merged**